tools/repl/repl.cpp

```diff
@@ -222,6 +227,38 @@ static bool needsAnotherLine(llvm::StringRef input) {
   return !stack.empty();
 }
 
+#if HAVE_LIBREADLINE
+// Load history file or create it
+std::error_code loadOrCreateHistoryFile(llvm::SmallString<128>& historyFile) {
+  llvm::Twine baseHistoryFile = llvm::Twine(HISTORY_FILE);
+
+  if (llvm::sys::path::home_directory(historyFile)) {
```

Comment:
nit: can you invert this check, so logic fails fast if there's no homedir?  Thanks!

Reply:
The purpose of line 235 is not to only to check if there is a home directory, but to get the location of the home directory, so I can append to the file name to the home directory location. Maybe I didn't understand your idea, but I didn't see how to do that.

Reply:
WhatI had in mind was to restructure the function body like this:
```C++
...
if (!llvm::sys::path::home_directory(historyFile)) {
  // Use ENOENT here since it could not found a home directory
  return std::error_code(ENOENT, std::system_category());
}

llvm::sys::path::append(historyFile, baseHistoryFile);
...
return std::error_code();

Reply:
Oh, that's great! I was actually feeling uncomfortable with this, and your idea fulfill exactly what I wanted. Thank you so much!

Reply:
Changed